File: official/docs/java/pickups/BuyPickup.java
import com.easypost.EasyPost;
import java.util.HashMap;

public class BuyPickup {
    public static void main(String[] args) {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");
        
        Pickup pickup = Pickup.retrieve("pickup_...");

        HashMap<String, Object> buyMap = new HashMap<String, Object>();
        buyMap.put("carrier", "UPS");
        buyMap.put("service", "Same-day Pickup");

        pickup = pickup.buy(buyMap);

        System.out.println(pickup);
    }
}

File: official/docs/php/pickups/BuyPickup.php
<?php

\EasyPost\EasyPost::setApiKey($_ENV['EASYPOST_API_KEY']);

$pickup = \EasyPost\Pickup::retrieve('pickup_...');

$pickup->buy([
    'carrier' => 'UPS',
    'service' => 'Same-day Pickup'
]);

echo $pickup;
